src/components/toolbar/toolbar.spec.js

```diff
@@ -15,74 +15,65 @@ describe('<md-toolbar>', function() {
-      mdShrinkSpeedFactor: 1,
-      $observe: function() {}
-    });
+    document.body.appendChild(element[0]);
 
-    $rootScope.$apply();
-    $rootScope.$broadcast('$mdContentLoaded', contentEl);
     $$rAF.flush();
 
+    var toolbarStyles = getComputedStyle(element.find('md-toolbar')[0]);
+    var contentStyles = getComputedStyle(element.find('md-content')[0]);
+    var toolbarHeight = element.find('md-toolbar').prop('offsetHeight');
+    var transformToolbar = getTransform(toolbarStyles);
+    var transformContent = getTransform(contentStyles);
 
-    //Expect everything to be in its proper initial state.
-    expect(toolbarCss[$mdConstant.CSS.TRANSFORM]).toEqual('translate3d(0,0px,0)');
-    expect(contentCss['margin-top']).toEqual('-100px');
-    expect(contentCss['margin-bottom']).toEqual('-100px');
-    expect(contentCss[$mdConstant.CSS.TRANSFORM]).toEqual('translate3d(0,100px,0)');
+    expect(transformToolbar).toEqual('matrix(1, 0, 0, 1, 0, 0)');
+    expect(contentStyles.marginTop).toEqual('-' + (toolbarHeight / 2) + 'px');
+    expect(contentStyles.marginBottom).toEqual((toolbarHeight / 2) + 'px');
+    expect(transformContent).toEqual('matrix(1, 0, 0, 1, 0, ' + (toolbarHeight / 2) + ')');
 
-    // Fake scroll to the bottom
-    contentEl.triggerHandler({
+    element.find('md-content').triggerHandler({
       type: 'scroll',
-      target: {scrollTop: 500}
+      target: {
+        scrollTop: 500
+      }
     });
     $$rAF.flush();
 
-    expect(toolbarCss[$mdConstant.CSS.TRANSFORM]).toEqual('translate3d(0,-100px,0)');
-    expect(contentCss[$mdConstant.CSS.TRANSFORM]).toEqual('translate3d(0,0px,0)');
+    toolbarStyles = getComputedStyle(element.find('md-toolbar')[0]);
+    contentStyles = getComputedStyle(element.find('md-content')[0]);
+    transformToolbar = getTransform(toolbarStyles);
+    transformContent = getTransform(contentStyles);
 
-    // Fake scroll back to the top
-    contentEl.triggerHandler({
+    expect(transformToolbar).toEqual('matrix(1, 0, 0, 1, 0, -' + toolbarHeight + ')');
+    expect(contentStyles.marginTop).toEqual('-' + (toolbarHeight / 2) + 'px');
+    expect(contentStyles.marginBottom).toEqual('-' + (toolbarHeight / 2) + 'px');
+    expect(transformContent).toEqual('matrix(1, 0, 0, 1, 0, -' + (toolbarHeight / 2) + ')');
+
+    element.find('md-content').triggerHandler({
       type: 'scroll',
-      target: {scrollTop: 0}
+      target: {
+        scrollTop: 0
+      }
     });
     $$rAF.flush();
 
-    expect(toolbarCss[$mdConstant.CSS.TRANSFORM]).toEqual('translate3d(0,0px,0)');
-    expect(contentCss[$mdConstant.CSS.TRANSFORM]).toEqual('translate3d(0,100px,0)');
+    toolbarStyles = getComputedStyle(element.find('md-toolbar')[0]);
+    contentStyles = getComputedStyle(element.find('md-content')[0]);
+    transformToolbar = getTransform(toolbarStyles);
+    transformContent = getTransform(contentStyles);
+
+    expect(transformToolbar).toEqual('matrix(1, 0, 0, 1, 0, 0)');
```

Comment:
Oh, it should work without matrix as well. Just as before, by using the `$mdConstant` and computed style. Matrix is not that fancy.